docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+# Runtime image
+
+## Overview
+
+For a dynamic languages like PHP, Python or Ruby, the build-time and run-time environments are the same. In this case using builder as a base image for a resulting image is natural.
+
+For compiled languages like C/C++, Java or Golang, the dependencies necessary for compilation might dramatically outweigh the actual runtime artifacts. To keep runtime images slim, S2I enables multiple-step build processes, where a binary artifact such as an executable or Java WAR file is created in the first builder image, extracted, and injected into an second image that simply places the executable in the correct location for execution. To give you even more abilities to customize a resulting image, S2I is also executing `assemble-runtime` script inside of this (runtime) image. In this way you may do a final adjustments by modifying files before an image will be committed.
+
+The following scheme illustrates the build workflow:
+
+![s2i workflow](./runtime-image-flow.png "s2i workflow")
+
+
+## How it works?
+
+To make this work S2I needs to know the following:
+
+* builder image
+* mapping of the artifacts
+* runtime image
+
+And all these information you should specify during S2I invocation:
+
+    s2i build <repo> <builder-image> <my-app> --runtime-image <runtime-image> --runtime-artifact </path/to/artifact>
+
+The only new here are `--runtime-image` and `--runtime-artifact` options. The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files. Note that you can specify this option multiple times.
```

Comment:
"The first option specifies the image that will be used as the base image for the final application image.  The second option specifies a full path to a file (or directory) that will exist in the builder image after assembly, and a corresponding directory on the runtime image where S2I will copy these files, separated by a colon(?).  Note that you can specify this option multiple times."

Also you need to update the example invocation to specify the destination path.  Or otherwise do a better job of explaining that the destination path is optional, and how to specify it if desired.